MapboxCoreNavigation/RouteController.swift

```diff
@@ -316,6 +333,35 @@ open class RouteController: NSObject {
     }
     
     /**
+     Checks whether a new location is quality by comparing it to the previous location.
+     
+     We're attempting to throw out coordinates that have a large difference in location or time.
+     */
+    func isQualityLocation(location: CLLocation?) -> Bool {
+        
+        // Any time we return in this func, reset the values.
+        defer {
+            if let location = location {
+                if let lastLocation = lastQualityLocationInterval {
+                    lastQualityLocationInterval?.speedDelta = location.speed - lastLocation.location.speed
+                }
+                
+                lastQualityLocationInterval?.location = location
+            }
+        }
+        
+        guard let lastLocation = lastQualityLocationInterval, let location = rawLocation else {
+                return false
+        }
+        
+        guard location.speed <= lastLocation.speedDelta * RouteControllerDifferenceInSpeedDeltaMultiplier else {
```

Comment:
It’s incorrect to compare a speed (measured in m/s) to a difference in speeds (measured in m/s² assuming location updates every second). We should either evaluate `abs(lastLocation.speed - location.speed)` (speed difference in m/s, or acceleration in m/s²) or compare `abs(lastLocation.speed - location.speed) / lastLocation.speed` (which would be a percentage). Either way, we should get rid of `speedDelta`.